datasets/nq_open/nq_open.py

```diff
@@ -0,0 +1,133 @@
+}
+"""
+
+_URLS = {
+    "dev": "https://raw.githubusercontent.com/google-research-datasets/natural-questions/master/nq_open/NQ-open.efficientqa.dev.1.1.jsonl",
+    "train": "https://raw.githubusercontent.com/google-research-datasets/natural-questions/master/nq_open/NQ-open.train.jsonl",
+}
+
+
+class NQOpenV2Config(datasets.BuilderConfig):
+    """BuilderConfig for NQ_Open."""
+
+    def __init__(self, **kwargs):
+        """BuilderConfig for NQ_Open.
+        Args:
+          **kwargs: keyword arguments forwarded to super.
+        """
+        super(NQOpenV2Config, self).__init__(**kwargs)
+
+
+class NQOpen(datasets.GeneratorBasedBuilder):
+    """ NQ_Open open domain question answering dataset. """
+
+    BUILDER_CONFIGS = [
+        NQOpenV2Config(
+            name="nq_open",
+            version=datasets.Version("1.0.0", ""),
+            description=f"NQ_Open open domain question answering dataset.",
+        ),
+    ]
+    BUILDER_CONFIG_CLASS = NQOpenV2Config
+
+    def _info(self):
+        return datasets.DatasetInfo(
+            description=_DESCRIPTION,
+            features=datasets.Features(
+                {
+                    "question": datasets.Value("string"),
+                    "answer": datasets.Sequence(datasets.Value("string")),
+                }
+            ),
+            supervised_keys=None,
+            homepage=_HOMEPAGE_URL,
+            citation=_CITATION,
+        )
+
+    def _split_generators(self, dl_manager):
+        train_path = dl_manager.download_and_extract(_URLS["train"])
+        val_path = dl_manager.download_and_extract(_URLS["dev"])
+
+        return [
+            datasets.SplitGenerator(
+                name=datasets.Split.TRAIN,
+                gen_kwargs={"datapath": train_path},
+            ),
+            datasets.SplitGenerator(
+                name=datasets.Split.VALIDATION,
+                gen_kwargs={"datapath": val_path},
+            ),
+        ]
+
+    def _generate_examples(self, datapath):
+        """Generate NQ_Open examples."""
+
+        with open(datapath, encoding="utf-8") as json_file:
+            json_list = list(json_file)
+
+        for i, json_str in enumerate(json_list):
+            result = json.loads(json_str)
+            yield i, result
```

Comment:
To avoid loading the whole jsonl file in memory at once, and to take full advantage of the generator pattern, I'd recommend the following syntax:
```suggestion
            for i, json_str in enumerate(json_file):
                result = json.loads(json_str)
                yield i, result
```